packages/cli/src/lib/agents/copilot.ts

```diff
@@ -0,0 +1,246 @@
+import { launch, launchSync } from 'rover-common';
+import {
+  AIAgentTool,
+  InvokeAIAgentError,
+  MissingAIAgentError,
+} from './index.js';
+import { PromptBuilder, IPromptTask } from '../prompts/index.js';
+import { parseJsonResponse } from '../../utils/json-parser.js';
+import { homedir } from 'node:os';
+import { join } from 'node:path';
+import { existsSync } from 'node:fs';
+
+// Environment variables reference:
+// - Run `copilot help environment` for official environment variables documentation
+const COPILOT_ENV_VARS = [
+  // GitHub Copilot CLI configuration
+  'COPILOT_ALLOW_ALL',
+  'COPILOT_CUSTOM_INSTRUCTIONS_DIRS',
+  'COPILOT_MODEL',
+  
+  // GitHub authentication
+  'GH_TOKEN',
+  'GITHUB_TOKEN',
+  
+  // General configuration
+  'XDG_CONFIG_HOME',
+  'XDG_STATE_HOME',
+  'NO_COLOR',
+  'CLICOLOR',
+  'CLICOLOR_FORCE',
+  'HTTP_PROXY',
+  'HTTPS_PROXY',
+  'DEBUG',
+];
+
+class CopilotAI implements AIAgentTool {
+  // constants
+  public AGENT_BIN = 'copilot';
+  private promptBuilder = new PromptBuilder('copilot');
+
+  constructor() {
+    // Check standalone Copilot CLI is available
+    try {
+      launchSync(this.AGENT_BIN, ['--version']);
+    } catch (err) {
+      throw new MissingAIAgentError(this.AGENT_BIN);
+    }
+  }
+
+  async invoke(prompt: string, json: boolean = false): Promise<string> {
+    let finalPrompt = prompt;
+    
+        if (json) {
+          // GitHub Copilot CLI is conversational and doesn't support structured JSON output.
+          // We need to be very explicit about JSON-only responses
+          finalPrompt = `${prompt}
+
+CRITICAL: You MUST respond with ONLY a valid JSON object. No conversational text, no explanations, no markdown, no bullet points, no greetings. Start your response with { and end with }. Nothing else.
```

Comment:
```suggestion
CRITICAL: You MUST respond with ONLY a valid JSON object. No conversational text, no explanations, no markdown, no bullet points, no greetings. Nothing else.
```

I'd just leave the room open for arrays to be returned toplevel, since they are also valid JSON expressions. If this makes the model to sometimes not return valid JSON then we can keep the prompt as it is right now.